src/composeappmanager.cc

```diff
@@ -64,15 +115,40 @@ ComposeAppConfig::ComposeAppConfig(const PackageConfig &pconfig) {
     boost::algorithm::to_lower(val);
     docker_prune = val != "0" && val != "false";
   }
+
+  // Docker Registry Configuration (TODO: to be removed)
```

Comment:
Can we just remove this stuff and hard-code it in `getBasicAuthHeader()` where we already have the big TODO's? I'm hoping this will never leak into someone's sota.toml. Or do you need someway to intercept things for unit tests?

Reply:
What about the on-premises deployment? In this case, the base URL will be different than 'https://ota-lite.foundries.io:8443'.

Reply:
ugh - your right. l guess we should keep the logic that discovers it - lets drop the config value method though.

Reply:
What is `the config value method`?
I guess you mean that we just deduce the auth base url from `ostree_server` url and append `hub-creds`  to it?

Reply:
correct. we deduce from ostree_server

Reply:
Removing the ability to configure the auth URL and just deducting it from the ostree/treehub URL is fine.
But, removing this param auth_creds_endpoint and moving the deduction logic in `getBasicAuthHeader` is not fine. First of all, we need to pass at least one param to RegistryClient anyway (either the treehub URL or already deducted auth URL), secondly if we do deduction in `getBasicAuthHeader` then we will do it for each request while this URL is static and should be evaluated once at startup.

Reply:
ack.

Reply:
done